pkg/kinflate/resource/appresource.go

```diff
@@ -17,37 +17,27 @@ limitations under the License.
 package resource
 
 import (
-	kutil "k8s.io/kubectl/pkg/kinflate/util"
 	"k8s.io/kubectl/pkg/loader"
 )
 
-func resourcesFromPath(loader loader.Loader, path string) ([]*Resource, error) {
+func resourcesFromPath(loader loader.Loader, path string) (ResourceCollection, error) {
 	content, err := loader.Load(path)
 	if err != nil {
 		return nil, err
 	}
 
-	objs, err := kutil.Decode(content)
-	if err != nil {
-		return nil, err
-	}
-
-	var res []*Resource
-	for _, obj := range objs {
-		res = append(res, &Resource{Data: obj})
-	}
-	return res, nil
+	return decodeToResourceCollection(content)
 }
 
 //  NewFromPaths returns a slice of Resources given a resource path slice from manifest file.
-func NewFromPaths(loader loader.Loader, paths []string) ([]*Resource, error) {
-	allResources := []*Resource{}
+func NewFromPaths(loader loader.Loader, paths []string) (ResourceCollection, error) {
```

Comment:
I remember some discussion about keeping them as slices, but don't remember the why now :)

Reply:
We wanted to get rid of `KObject` at first so we kept it as slices. When working on the code for Application interfaces, we realized that we don't achieve it without a map from GVKN to unstructured. So we plan to rename `KObject` to `ResourceCollection`.